Building Grub fails when using gcc9

## Bug description

Building Grub using the `boards/x86/common/scripts/build_grub.sh` on a system with gcc9 (observed on Clear Linux 29750, gcc 9.1.1) installed will fail with the following error:
```
cc -DHAVE_CONFIG_H -I.  -Wall -W -DGRUB_UTIL=1 -D_FILE_OFFSET_BITS=64 -I./include -DGRUB_FILE=\"grub-core/fs/cpio.c\" -I. -I. -I. -I. -I./include -I./include -I./grub-core/lib/libgcrypt-grub/src/  -I./grub-core/lib/minilzo -I./grub-core/lib/xzembed -I./grub-core/lib/zstd -DMINILZO_HAVE_CONFIG_H -D_FILE_OFFSET_BITS=64 -Wall -W -Wshadow -Wpointer-arith -Wundef -Wchar-subscripts -Wcomment -Wdeprecated-declarations -Wdisabled-optimization -Wdiv-by-zero -Wfloat-equal -Wformat-extra-args -Wformat-security -Wformat-y2k -Wimplicit -Wimplicit-function-declaration -Wimplicit-int -Wmain -Wmissing-braces -Wmissing-format-attribute -Wmultichar -Wparentheses -Wreturn-type -Wsequence-point -Wshadow -Wsign-compare -Wswitch -Wtrigraphs -Wunknown-pragmas -Wunused -Wunused-function -Wunused-label -Wunused-parameter -Wunused-value  -Wunused-variable -Wwrite-strings -Wnested-externs -Wstrict-prototypes -Wcast-align  -Wextra -Wattributes -Wendif-labels -Winit-self -Wint-to-pointer-cast -Winvalid-pch -Wmissing-field-initializers -Wnonnull -Woverflow -Wvla -Wpointer-to-int-cast -Wstrict-aliasing -Wvariadic-macros -Wvolatile-register-var -Wpointer-sign -Wmissing-include-dirs -Wmissing-prototypes -Wmissing-declarations -Wformat=2 -Werror  -fno-builtin -Wno-undef  -MT grub-core/fs/libgrubmods_a-cpio.o -MD -MP -MF grub-core/fs/.deps-util/libgrubmods_a-cpio.Tpo -c -o grub-core/fs/libgrubmods_a-cpio.o `test -f 'grub-core/fs/cpio.c' || echo './'`grub-core/fs/cpio.c
In file included from grub-core/fs/cpio.c:51:
grub-core/fs/cpio_common.c: In function ‘grub_cpio_find_file’:
grub-core/fs/cpio_common.c:58:31: error: taking address of packed member of ‘struct head’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
   58 |   data->size = read_number (hd.filesize, ARRAY_SIZE (hd.filesize));
      |                             ~~^~~~~~~~~
grub-core/fs/cpio_common.c:60:29: error: taking address of packed member of ‘struct head’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
   60 |     *mtime = read_number (hd.mtime, ARRAY_SIZE (hd.mtime));
      |                           ~~^~~~~~
grub-core/fs/cpio_common.c:61:28: error: taking address of packed member of ‘struct head’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
   61 |   modeval = read_number (hd.mode, ARRAY_SIZE (hd.mode));
      |                          ~~^~~~~
grub-core/fs/cpio_common.c:62:29: error: taking address of packed member of ‘struct head’ may result in an unaligned pointer value [-Werror=address-of-packed-member]
   62 |   namesize = read_number (hd.namesize, ARRAY_SIZE (hd.namesize));
      |                           ~~^~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [Makefile:7120: grub-core/fs/libgrubmods_a-cpio.o] Error 1
make[2]: *** Waiting for unfinished jobs....
mv -f grub-core/fs/.deps-util/libgrubmods_a-afs.Tpo grub-core/fs/.deps-util/libgrubmods_a-afs.Po
mv -f grub-core/fs/.deps-util/libgrubmods_a-cbfs.Tpo grub-core/fs/.deps-util/libgrubmods_a-cbfs.Po
mv -f grub-core/fs/.deps-util/libgrubmods_a-bfs.Tpo grub-core/fs/.deps-util/libgrubmods_a-bfs.Po
mv -f grub-core/fs/.deps-util/libgrubmods_a-btrfs.Tpo grub-core/fs/.deps-util/libgrubmods_a-btrfs.Po
make[2]: Leaving directory '/home/gvancuts/zephyrproject/zephyr/boards/x86/common/scripts/grub/src'
make[1]: *** [Makefile:11589: all-recursive] Error 1
make[1]: Leaving directory '/home/gvancuts/zephyrproject/zephyr/boards/x86/common/scripts/grub/src'
make: *** [Makefile:3622: all] Error 2
./boards/x86/common/scripts/build_grub.sh: line 40: ./grub-mkimage: No such file or directory
~/zephyrproject/zephyr/boards/x86/common/scripts/grub ~/zephyrproject/zephyr
cp: cannot stat 'src/grub_x86_64.efi': No such file or directory
```
## How to rproduce
Steps to reproduce the behavior:
1. Run `boards/x86/common/scripts/build_grub.sh x86_64`
2. See error pasted above

## Impact
It is not possible to build the Grub bootloader, used to boot Zephyr as a UOS on ACRN for example (see https://docs.zephyrproject.org/latest/boards/x86/acrn/doc/index.html)

--- a/boards/x86/common/scripts/build_grub.sh
+++ b/boards/x86/common/scripts/build_grub.sh
@@ -22,7 +22,7 @@ prepare() {
   fi
 
   pushd src
-  git checkout grub-2.02-285-g5bc41db75
+  git checkout grub-2.04-rc1-17-g8e8723a6b
   git clean -fdx
   popd
 }

